zerver/webhooks/bitbucket3/view.py

```diff
@@ -0,0 +1,173 @@
+    event_type = change["type"]
+    repo_name = payload["repository"]["name"]
+    tag_name = change["ref"]["displayId"]
+    user_name = payload["actor"]["name"]
+
+    if event_type == "ADD":
+        action = "pushed"
+    elif event_type == "DELETE":
+        action = "removed"
+    else:
+        raise UnexpectedWebhookEventType("BitBucket3", event_type)
+
+    subject = BITBUCKET_TOPIC_TEMPLATE.format(repository_name=repo_name)
+    body = get_push_tag_event_message(
+        user_name,
+        tag_name,
+        action=action)
+    return {"subject": subject, "body": body}
+
+def repo_push_handler(payload: Dict[str, Any], branches: Optional[str]=None) -> List[Dict[str, str]]:
+    data = []
+    for change in payload["changes"]:
+        event_target_type = change["ref"]["type"]
+        if event_target_type == "BRANCH":
+            branch = change['ref']['displayId']
+            if branches:
+                if branch not in branches:
+                    continue
+            data.append(repo_push_branch_data(payload, change))
+        elif event_target_type == "TAG":
+            data.append(repo_push_tag_data(payload, change))
+        else:
+            raise UnexpectedWebhookEventType("BitBucket3", event_target_type)
+    return data
+
+EVENT_HANDLER_MAP = {
+    "repo": {
+        "comment": {
+            "added": partial(repo_comment_handler, action="commented"),
+            "edited": partial(repo_comment_handler, action="edited their comment"),
+            "deleted": partial(repo_comment_handler, action="deleted their comment"),
+        },
+        "forked": repo_forked_handler,
+        "modified": repo_modified_handler,
+        "refs_changed": repo_push_handler
+    },
+    "pr": {
+        "comment": {
+            "added": None,
+            "edited": None,
+            "deleted": None,
+        },
+        "declined": None,
+        "deleted": None,
+        "merged": None,
+        "modified": None,
+        "opened": None,
+        "reviewer": {
+            "approved": None,
+            "needs_work": None,
+            "updated": None,
+            "unapproved": None,
+        }
+    }
+}
+
+def get_handler_from_eventkey(eventkey: str) -> Callable[..., List[Dict[str, Any]]]:
+    map = EVENT_HANDLER_MAP  # type: Any
+    details = eventkey.split(":")
+    for detail in details:
+        map = map.get(detail)
+        if map is None:
+            break  # nocoverage
+    return map
+
+@api_key_only_webhook_view("Bitbucket3")
+@has_request_variables
+def api_bitbucket3_webhook(request: HttpRequest, user_profile: UserProfile,
+                           payload: Dict[str, Any]=REQ(argument_type="body"),
+                           branches: Optional[str]=REQ(default=None),
+                           user_specified_topic: Optional[str]=REQ("topic", default=None)) -> HttpResponse:
+    eventkey = payload.get("eventKey", "None")
+    handler = get_handler_from_eventkey(eventkey)
+    if not handler:
```

Comment:
You should check for `None` explicitly here. As in `if handler is None` :)

Reply:
A syntactical change? Sure thing!